.goreleaser.yml

```diff
@@ -0,0 +1,44 @@
+archives:
+  - files:
+      - none*
+    format: zip
+    name_template: '{{ .ProjectName }}_{{ .Version }}_{{ .Os }}_{{ .Arch }}'
+before:
+  hooks:
+    - go mod download
+builds:
+  - binary: '{{ .ProjectName }}_{{ .Version }}'
+    goarch:
+      - '386'
+      - amd64
+      - arm
+      - arm64
+    goos:
+      - darwin
+      - freebsd
+      - linux
+      - windows
+    ignore:
+      - goarch: '386'
+        goos: darwin
+    ldflags:
+      - -s -w -X github.com/terraform-providers/terraform-provider-aws/aws/version.ProviderVersion={{.Version}}
```

Comment:
You are probably going to want `-trimpath` for `flags` here as well as overriding the `mod_timestamp`, see these very helpful instructions an awesome contributor put together for Goreleaser: https://goreleaser.com/customization/build/#reproducible-builds

Reply:
Is https://github.com/terraform-providers/terraform-provider-aws/pull/14033/commits/aa78000afa74380bb785fae709c8a4559a9d2f49 what you had in mind here? 😄

Reply:
Yep, I can't really speak to the ldflags -X change, but the other parts look good.